scripts/build_with_EESSI_install.slurm

```diff
@@ -0,0 +1,95 @@
+    exit 3
+fi
+
+if [ x"${BUILD_TARGET_ARCHITECTURE}" == "x" ]; then
+    echo "variable BUILD_TARGET_ARCHITECTURE is not set; provide it via 'app.cfg'"
+    exit 4
+fi
+
+if [ x"${BUILD_EESSI_VERSION}" == "x" ]; then
+    echo "variable BUILD_EESSI_VERSION is not set; provide it via 'app.cfg'"
+    exit 5
+fi
+
+env | egrep 'SOFTWARE_FILE|SCRIPTS_DIR|LOCAL_TMP|BUILD_TARGET_ARCHITECTURE|BUILD_EESSI_VERSION'
+
+# requires some stuff from https://github.com/EESSI/software-layer
+# these and other scripts of the bot should be made available in some directory or in
+# some directories and the locations should then be configured in some cfg file, e.g., app.cfg
+#   init/minimal_eessi_env --> stored under $SCRIPTS_DIR/init
+#   init/eessi_environment_variables --> stored under $SCRIPTS_DIR/init
+#   init/eessi_softwre_subdir_for_host.py --> stored under $SCRIPTS_DIR/init
+#   configure_easybuild --> stored under $SCRIPTS_DIR
+#   eessi_software_subdir.py --> stored under $SCRIPTS_DIR
+
+
+# setup environment variables for build container
+#   mostly what standard build environment contains with the additional bind mount
+#   for the job directory assumed to be the current directory $(pwd) which is mapped to
+#   /eessi_bot_job inside the container
+export EESSI_TMPDIR=${LOCAL_TMP}/${USER}/EESSI
+export SINGULARITY_CACHEDIR=${EESSI_TMPDIR}/singularity_cache
+export SINGULARITY_BIND="${EESSI_TMPDIR}/var-run-cvmfs:/var/run/cvmfs,${EESSI_TMPDIR}/var-lib-cvmfs:/var/lib/cvmfs,$(pwd):/eessi_bot_job"
+export SINGULARITY_HOME="${EESSI_TMPDIR}/home:/home/${USER}"
+export EESSI_PILOT_READONLY="container:cvmfs2 pilot.eessi-hpc.org /cvmfs_ro/pilot.eessi-hpc.org"
+export EESSI_PILOT_WRITABLE_OVERLAY="container:fuse-overlayfs -o lowerdir=/cvmfs_ro/pilot.eessi-hpc.org -o upperdir=${EESSI_TMPDIR}/overlay-upper -o workdir=${EESSI_TMPDIR}/overlay-work /cvmfs/pilot.eessi-hpc.org"
+
+# create directories needed by the container
+mkdir -p ${EESSI_TMPDIR}/{home,overlay-upper,overlay-work,var-lib-cvmfs,var-run-cvmfs}
+
+# save some environment settings we may need after entering the Gentoo Prefix
+#   softwarelist.yaml could be stored directly by EESSI bot into some job definition file
+echo "export SOFTWARE_FILE=${SOFTWARE_FILE}" > _env
+#   scripts dir could be a directory known through app.cfg or stored in job definition file
+echo "export SCRIPTS_DIR=${SCRIPTS_DIR}" >> _env
+#   only used to set EESSI_TMPDIR, could be taken from a job definition file
+echo "export LOCAL_TMP=${LOCAL_TMP}" >> _env
+#   probably not needed (because all information can be derived via scripts on build node, i.e.,
+#   inside a job); same as "${EESSI_OS_TYPE}/${EESSI_SOFTWARE_SUBDIR}"
+echo "export BUILD_TARGET_ARCHITECTURE=${BUILD_TARGET_ARCHITECTURE}" >> _env
+#   could be delivered as argument, only used to set EESSI_PILOT_VERSION, the latter might be
+#   set directly instead
+echo "export BUILD_EESSI_VERSION=${BUILD_EESSI_VERSION}" >> _env
```

Comment:
Why not just pass these variables as arguments to the `mod_EESSI_install.sh` script instead?

Maybe leave it as is for now, since I suspect most of this will no longer be required if we change tactics to just run an `eessi-bot-build.sh` script from the software-layer...

Reply:
Originally thought that I could just set these as environment variables by the bot before submitting the job. Only later I figured out that running in the combat layer cleans up all environment settings.

Agree to leave it and revise once we have the more generic approach (via `eessi-bot-build.sh`) in place.

Reply:
Created issue #8 to keep this in mind.